include/discpp/command_handler.h

```diff
@@ -6,11 +6,31 @@
 #include "message.h"
 #include "command.h"
 
-#include <memory>
+#include <type_traits>
 
 namespace discpp {
-	inline std::unordered_map<std::string, Command*> registered_commands;
+    class CommandHandler {
+        Client &client;
+        friend class Client;
+        friend class Command;
+      public:
+        std::unordered_map<std::string, Command *> registered_commands;
 
+        CommandHandler(Client &parent) : client(parent) {}
+
+        template <typename T, typename... Args/*, std::enable_if_t<std::is_base_of<Command, T>::value || std::is_same<T, Command>::value> * = nullptr*/>
+        void RegisterCommand(Args &&... args)
+        {
+            auto command = new T(args...);
```

Comment:
std::unique_ptr

Reply:
Thought the same but this library uses a lot of raw pointers so I stuck with it, I may update it later